worker/caasadmission/doc.go

```diff
@@ -0,0 +1,12 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package caasadmission worker is responsible for establishing a Kubernetes
+// admission webhook so that Juju can process CRUD based Kubernetes object
+// events. We currently use these admission webhooks to watch for resources
+// being created by charms in the Kubernetes controller. When we detect charms
+// creating Kubernetes resources, we add labels to the new or updated Kubernetes
+// objects, to indicate the Juju application that was responsible for creating
+// the resource. When the application is later removed, we use these labels to
+// find associated resources which need to be removed.
+package caasadmission
```

Comment:
Looks good, but here's a small suggestion for the first sentence, and a couple of less important ones for the rest (feel free to reject them if they alter the meaning). 

> // Package caasadmission defines the caasadmission worker. This worker is responsible 
// for establishing a Kubernetes admission webhook so that Juju can process CRUD-based 
// Kubernetes object events. We currently use these admission webhooks to watch 
// for resources being created by charms in the Kubernetes controller. Every time we detect 
// such a resource, we annotate the new or updated Kubernetes object with a label
// indicating which Juju application was responsible for creating the resource. 
// When the application is later removed, we use these labels to
// find associated resources that need to be removed.
package caasadmission

Also, a few questions:

(1) Where is the webhook being established?
(2) What's the connection between a resource and a Kubernetes object?
(3) Who creates the resource, the charm or the application delivered by the charm? We say both.
(4) Are there any other reasons why we keep track of which charm/application has created which resource, or is it always about tying the resource to the lifecycle of the charm/application so we can remove it automatically when the application is removed?

Reply:
I have updated the paragraph. To answer the questions:
1) It's created as part of our model creation process in Kubernetes.
2) An object is an instantiated resource in Kubernetes. I don't think we need to go into this level of detail.
3) The charm creates the resource but I have avoided using that language as you can have multiple of the same charmed deployed all creating their own resources in Kubernetes. Except the authority for which they are working under is the Juju application and that is how we track ownership. So for documentation purposes it is the Juju application that is creating the resources.
4) Just this reason at the moment. This may grow in the future but it has a very small foot print for now.